torch/csrc/jit/codegen/cuda/interface.cpp

```diff
@@ -34,11 +39,129 @@ void fuseGraph(std::shared_ptr<Graph>& graph) {
   getFuserInterface()->fn_fuse_graph(graph);
 }
 
+// [ Note -- type guard logic in CudaFusionGuard ]
```

Comment:
to the readers not familiar with this part of the code, this comment is confusing: what is it really saying? :)